scripts/linters/pylint_extensions.py

```diff
@@ -1653,100 +1653,121 @@ class SingleLineCommentChecker(checkers.BaseChecker):
-                if last_char_is_invalid:
-                    self.add_message(
-                        'invalid-punctuation-used', line=line_num + 1)
+    def process_tokens(self, tokens):
+        """Custom pylint checker to ensure that comments follow correct style.
 
-            # Comments must start with a space.
-            if re.search(br'^#[^\s].*$', line) and not line.startswith(b'#!'):
-                space_at_beginning_of_comment = False
-                self.add_message(
-                    'no-space-at-beginning', line=line_num + 1)
-
-            # Comments may include a lowercase character at the beginning
-            # only if they start with version info or a data type or.
-            # a variable name e.g. "# next_line is of string type."
-            # or "# v2 version does not have ExplorationStatsModel."
-            # or "# int. The file size, in bytes.".
-            if len(line) > 1 and space_at_beginning_of_comment:
-
-                # Check if variable name is used.
-                underscore_is_present = '_' in line.split()[1]
-                if underscore_is_present:
-                    continue
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+        """
+        prev_line_num = -1
+        comments = []
+        comments_index = -1
 
-                # Check if allowed prefix is used.
-                allowed_prefix_is_present = any(
-                    line[2:].startswith(word) for word in (
-                        allowed_comment_prefixes))
-                if allowed_prefix_is_present:
-                    continue
+        for (token_type, _, (line_num, _), _, line) in tokens:
+            if token_type == tokenize.COMMENT:
+                line = line.strip()
 
-            # Comments must start with a capital letter.
-            if not previous_line.startswith(b'#') and (
-                    re.search(br'^# [a-z][A-Za-z]*.*$', line)):
-                self.add_message(
-                    'no-capital-letter-at-beginning', line=line_num + 1)
+                if line.startswith('#'):
+                    if prev_line_num + 1 == line_num:
+                        comments[comments_index].append((line_num, line))
+                    else:
+                        comments.append([(line_num, line)])
+                        comments_index += 1
+                    prev_line_num = line_num
+
+        for comment in comments:
+            space_at_beginning_of_first_comment = True
+            space_at_beginning_of_last_comment = True
+            first_comment_line_num, first_comment = comment[0]
+            last_comment_line_num, last_comment = comment[-1]
+            space_at_beginning_of_first_comment = (
+                self._check_space_at_beginning_of_comment(
+                    first_comment, first_comment_line_num))
+            space_at_beginning_of_last_comment = (
+                self._check_space_at_beginning_of_comment(
```

Comment:
Are we expecting to check space at the beginning of each comment?

Reply:
Fixed the check.